docs/architecture/data.md

```diff
@@ -111,3 +113,46 @@ class first_gcs,second_gcs,validated_gcs gcs_group_boxstyle
 class first_bq,second_bq bq_group_boxstyle
 class gcs_label1,gcs_label2,gcs_label3,bq_label1,bq_label2 group_labelstyle
 ```
+
+## Adding a new data source
+
+Adding a new data source based on the architecture described above involves several steps, outlined below.
+
+```{note}
+If you're bringing in data that is similar to existing data (for example, a new subset of an existing dataset like a new Airtable or Littlepay table), you should follow the existing pattern for that dataset. [This spreadsheet](https://docs.google.com/spreadsheets/d/1bv1K5lZMnq1eCSZRy3sPd3MgbdyghrMl4u8HvjNjWPw/edit#gid=0) gives overviews of some prominent existing data sources, outlining the specific code/resources that correspond to each step in the [general data flow](data-ingest-diagram) for that data source.
+```
+
+### Determine upstream source type
+
+To determine the best storage location for your raw data (especially if it requires manual curation), consult the [Data Collection and Storage Guidance within the Cal-ITP Data Pipeline Google Doc](https://docs.google.com/document/d/1-l6c99UUZ0o3Ln9S_CAt7iitGHvriewWhKDftESE2Dw/edit).
+
+The [Should it be a dbt model?](tool_choice) docs section also has some guidance about when a data pipeline should be created.
+
+### Bring data into Google Cloud Storage
+
+We store our raw, un-transformed data in Google Cloud Storage to ensure that we can always recover the raw data if needed.
```

Comment:
Might be worth adding something like ", usually in perpetuity, " here to drive home the recoverability motivation